### PR TITLE
Snow 1223142 add like to list commands

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Add `--like` flag to `list-branches` and `list-tags` in `snow git` commands and `--pattern` flag to `list-files`

### CODE REVIEW

File: src/snowflake/cli/plugins/git/commands.py [hunk 1]
```diff
@@ -42,6 +42,24 @@ def _repo_path_argument_callback(path):
 )
 
 
+def like_option(help_example: str):
+    return typer.Option(
+        "%%",
+        "--like",
+        "-l",
+        help=f"SQL LIKE pattern for filtering objects by name. For example, {help_example}",
+    )
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/git/commands.py:51]: Why not use `snowflake.cli.api.commands.flags.LikeOption`?

File: src/snowflake/cli/plugins/git/manager.py [hunk 2]
```diff
@@ -3,16 +3,16 @@
...
 
-    def show_tags(self, repo_name: str) -> SnowflakeCursor:
-        query = f"show git tags in {repo_name}"
+    def show_tags(self, repo_name: str, like: str) -> SnowflakeCursor:
+        query = f"show git tags like '{like}' in {repo_name}"
         return self._execute_query(query)
 
-    def show_files(self, repo_path: str) -> SnowflakeCursor:
-        query = f"ls {repo_path}"
+    def show_files(self, repo_path: str, pattern: str) -> SnowflakeCursor:
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/git/manager.py:14]: How different is this from `StageManager.list_files`? Why not extend and use the parent method?

Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/git/manager.py]: Done with addition of
https://github.com/snowflakedb/snowflake-cli/pull/888

File: src/snowflake/cli/plugins/git/commands.py (hunk 1 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/git/commands.py]: Refactor with https://github.com/snowflakedb/snowflake-cli/pull/889

**Review by sfc-gh-turbaszek — APPROVED** [review]

File: src/snowflake/cli/plugins/git/commands.py (hunk 1 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/git/commands.py]: [merged]

**Pull request merged**